llvm/lib/Transforms/InstCombine/InstCombineSimplifyDemanded.cpp

```diff
@@ -420,6 +420,15 @@ Value *InstCombinerImpl::SimplifyDemandedUseBits(Value *V, APInt DemandedMask,
     break;
   }
   case Instruction::Trunc: {
+    unsigned SrcBitWidth = I->getOperand(0)->getType()->getScalarSizeInBits();
+    APInt InputDemandedMask = DemandedMask.zextOrTrunc(SrcBitWidth);
+    KnownBits InputKnown(SrcBitWidth);
+    if (SimplifyDemandedBits(I, 0, InputDemandedMask, InputKnown, Depth + 1)) {
+      // We may drop the nowrap flags
+      I->dropPoisonGeneratingFlags();
+      return I;
+    }
```

Comment:
This addition doesn't make sense to me -- we need to drop flags on existing transforms, not add a new one. But we actually already *do* drop flags because we fall through to zext below, which already has poison-flag handling for nneg. So actually everything is fine here without changes, and just the test addition is enough.